corehq/apps/app_manager/dbaccessors.py

```diff
@@ -52,6 +52,22 @@ def get_latest_released_app(domain, app_id):
     return None
 
 
+def get_latest_released_app_versions_by_app_id(domain):
+    """
+    Gets a dict of all apps in domain that have released at least one build
+    and the version of their most recently released build. Note that keys
+    are the app ids, not build ids.
+    """
+    from .models import Application
+    results = Application.get_db().view(
+        'app_manager/applications',
+        startkey=['^ReleasedApplications', domain],
+        endkey=['^ReleasedApplications', domain, {}],
+        include_docs=False,
+    ).all()
+    return {r['key'][2]: r['key'][3] for r in results}
```

Comment:
is it worth a note here saying that the view is ordered by app_id, version in ascending order so dict will end up with highest version number for each app?

Reply:
what makes it ordered by app_id, version?

Reply:
couch views are ordered by they key

Reply:
089cd8a5cfa